demo/image_demo.py

```diff
@@ -31,29 +78,31 @@ def main(args):
     # register all modules in mmdet into the registries
     register_all_modules()
 
-    # TODO: Support inference of image directory.
     # build the model from a config file and a checkpoint file
     model = init_detector(args.config, args.checkpoint, device=args.device)
 
     # init visualizer
     visualizer = VISUALIZERS.build(model.cfg.visualizer)
     visualizer.dataset_meta = model.dataset_meta
 
-    # test a single image
-    result = inference_detector(model, args.img)
-
-    # show the results
-    img = mmcv.imread(args.img)
-    img = mmcv.imconvert(img, 'bgr', 'rgb')
-    visualizer.add_datasample(
-        'result',
-        img,
-        data_sample=result,
-        draw_gt=False,
-        show=True,
-        wait_time=0,
-        out_file=args.out_file,
-        pred_score_thr=args.score_thr)
+    files = LoadFiles(args.img)
+    progress_bar = ProgressBar(len(files))
+
+    for file in files:
+        result = inference_detector(model, file)
+        img = mmcv.imread(file)
+        img = mmcv.imconvert(img, 'bgr', 'rgb')
+        _, file_name = os.path.split(file)
```

Comment:
Consider the case of the same file name. To avoid this, it is recommended to use the folder path as the file name.

a/b/c/d.jpg  -> a_b_c_d.jpg
a/b/d/d.jpg  -> a_b_d_d.jpg